File: app/models/property_list.rb
class PropertyList < ApplicationRecord
  validates :quality, presence: true
  validates :product_name, :sku, :brand, presence: true, format: { with: /\A[A-Za-z0-9_\-. ]+\z/, message: "Sallitut merkit: A-Z, 0-9, _-." }

  validates :listed, inclusion: { in: [true, false] }, exclusion: { in: [nil] }

  belongs_to :product, polymorphic: true
  validates :product, presence: true
end

File: pcgen-front/app/models/property-list.js
import DS from 'ember-data';

export default DS.Model.extend({
  listed: DS.attr('boolean'),
  quality: DS.attr('number'),
  productName: DS.attr('string'),
  sku: DS.attr('string'),
  brand: DS.attr('string'),
  pricesUrl: DS.attr('string'),
  specsUrl: DS.attr('string'),
  description: DS.attr('string'),
  warranty: DS.attr('number'),
  lengthMm: DS.attr('number'),
  widthMm: DS.attr('number'),
  heightMm: DS.attr('number'),
  colors: DS.attr('string'),
  ledColor: DS.attr('string'),
  product: DS.belongsTo('product', { polymorphic: true })
});
